packages/salesforcedx-vscode-core/test/vscode-integration/providers/configAggregatorProvider.test.ts

```diff
@@ -0,0 +1,205 @@
+        (ConfigAggregatorProvider as any).prototype,
+        'getCurrentDirectory'
+      );
+      configAggregatorCreateSpy = sandbox.spy(ConfigAggregator, 'create');
+    });
+
+    it('should create a ConfigAggregator when outside of a project directory', async () => {
+      // Arrange
+      getCurrentDirectoryStub
+        .onCall(0)
+        .returns(ConfigAggregatorProvider.defaultBaseProcessDirectoryInVSCE);
+
+      // Act
+      const configAggregator = await (configAggregatorProvider as any).createConfigAggregator();
+
+      // Assert
+      // createConfigAggregator should store the cwd initially,
+      // and check it again after creating the ConfigAggregator
+      // to ensure that the cwd is set back to its original value.
+      expect(getCurrentDirectoryStub.callCount).to.equal(2);
+      // createConfigAggregator should change the process to the
+      // root project workspace path to create the ConfigAggregator,
+      // then change back to the original directory.
+      expect(changeCurrentDirectoryToStub.callCount).to.equal(2);
+      expect(changeCurrentDirectoryToStub.getCall(1).args[0]).to.equal(
+        ConfigAggregatorProvider.defaultBaseProcessDirectoryInVSCE
+      );
+      expect(configAggregatorCreateSpy.callCount).to.equal(1);
+      expect(configAggregator).to.not.equal(undefined);
+    });
+
+    it('should create a ConfigAggregator when inside of a project directory', async () => {
+      // Arrange
+      getCurrentDirectoryStub.returns(dummyProjectRootWorkspacePath);
+      getRootWorkspacePathStub.returns(dummyProjectRootWorkspacePath);
+
+      // Act
+      const configAggregator = await (configAggregatorProvider as any).createConfigAggregator();
+
+      // Assert
+      // createConfigAggregator should store the cwd initially,
+      // and check it again after creating the ConfigAggregator
+      // to ensure that the cwd is set back to its original value.
+      expect(getCurrentDirectoryStub.callCount).to.equal(2);
+      // Since the stubbed current directory is the same as the root
+      // workspace path, the directory should not have been changed.
+      expect(changeCurrentDirectoryToStub.callCount).to.equal(0);
+      expect(configAggregatorCreateSpy.callCount).to.equal(1);
+      expect(configAggregator).to.not.equal(undefined);
+      expect(getRootWorkspacePathStub.callCount).to.equal(1);
+    });
+
+    it('should create an SfdxConfigAggregator', async () => {
+      // Arrange
+      getCurrentDirectoryStub
+        .onCall(0)
+        .returns(ConfigAggregatorProvider.defaultBaseProcessDirectoryInVSCE);
+      const sfdxConfigAggregatorCreateSpy = sandbox.spy(
+        SfdxConfigAggregator,
+        'create'
+      );
+
+      // Act
+      const sfdxConfigAggregator = await (configAggregatorProvider as any).createConfigAggregator(
+        { sfdx: true }
+      );
+
+      // Assert
+      expect(sfdxConfigAggregatorCreateSpy.callCount).to.equal(1);
```

Comment:
should we assert here that the regular create method was not called?

Reply:
Removed since no need for sfdx flavor of Config Aggregator anymore - the new ConfigAgg works with the old key